ouroboros-consensus/ouroboros-consensus-mock/src/Ouroboros/Consensus/Mock/Ledger/UTxO.hs

```diff
@@ -89,61 +104,63 @@ data UtxoError
 instance Condense UtxoError where
   condense = show
 
-class HasUtxo a where
-  txIns      :: a -> Set TxIn
-  txOuts     :: a -> Utxo
-  confirmed  :: a -> Set TxId
-  updateUtxo :: a -> Utxo -> Except UtxoError Utxo
-
-{-------------------------------------------------------------------------------
-  HasUtxo instances
--------------------------------------------------------------------------------}
-
-instance HasUtxo Tx where
-  txIns     (Tx ins _outs) = ins
-  txOuts tx@(Tx _ins outs) =
-      Map.fromList $ zipWith aux [0..] outs
-    where
-      aux :: Ix -> TxOut -> (TxIn, TxOut)
-      aux ix out = ((hash tx, ix), out)
-
-  confirmed       = Set.singleton . hash
-  updateUtxo tx = execStateT $ do
-    -- Remove all inputs from the Utxo and calculate the sum of all the input
-    -- amounts
-    inputAmount <- fmap sum $ forM (Set.toList (txIns tx)) $ \txIn -> do
-      u <- get
-      case Map.updateLookupWithKey (\_ _ -> Nothing) txIn u of
-        (Nothing,              _)  -> throwError $ MissingInput txIn
-        (Just (_addr, amount), u') -> put u' $> amount
-
-    -- Check that the sum of the inputs is equal to the sum of the outputs
-    let outputAmount = sum $ map snd $ Map.elems $ txOuts tx
-    when (inputAmount /= outputAmount) $
-      throwError $ InputOutputMismatch inputAmount outputAmount
-
-    -- Add the outputs to the Utxo
-    modify (`Map.union` txOuts tx)
-
-instance HasUtxo a => HasUtxo [a] where
-  txIns      = foldr (Set.union . txIns)     Set.empty
-  txOuts     = foldr (Map.union . txOuts)    Map.empty
-  confirmed  = foldr (Set.union . confirmed) Set.empty
-  updateUtxo = repeatedlyM updateUtxo
-
-instance HasUtxo a => HasUtxo (Chain a) where
-  txIns      = txIns      . toOldestFirst
-  txOuts     = txOuts     . toOldestFirst
-  updateUtxo = updateUtxo . toOldestFirst
-  confirmed  = confirmed  . toOldestFirst
+class HasTxs a where
```

Comment:
There already exists such a class: https://github.com/input-output-hk/ouroboros-network/blob/cca25adb2cc393177286feaba4c503a033343c36/ouroboros-consensus/src/Ouroboros/Consensus/Mempool/API.hs#L93

Either we reuse that class (maybe generalise it to `a` instead of `blk`), and if that's not possible (`Tx` vs `GenTx`), rename this new class to avoid confusion.

Reply:
I renamed the new one; the two have slightly different semantics (going by the existing class's comment). Moreover, this new one is specific to the mock Tx -- I don't think generalizing the existing one would be trivial and/or as clear.

Done, thanks.